src/app/triggerFromForm.ts

```diff
@@ -0,0 +1,124 @@
+import {Tickets} from 'node-zendesk';
+
+import {AppContext, AppCallValues} from 'mattermost-redux/types/apps';
+
+import {SubscriptionFields, TriggerFields} from '../utils/constants';
+import {FieldValidationErrors} from '../utils/call_responses';
+
+interface ITriggerFromFrom {
+    getTrigger(): Tickets.CreatePayload;
+    fieldValidationErrors: FieldValidationErrors;
+}
+
+export class TriggerFromForm implements ITriggerFromFrom {
+    values: AppCallValues;
+    context: AppContext;
+    trigger: any
+    fieldValidationErrors: FieldValidationErrors
+
+    constructor(values: AppCallValues, context: AppContext) {
+        this.values = values;
+        this.context = context;
+        this.fieldValidationErrors = {};
+        this.trigger = {};
+    }
+
+    getTrigger(): any {
+        // If not a new trigger, add the trigger ID to the payload
+        // This is a signal to update the trigger, not create a new one
+        if (!this.isNewTrigger()) {
+            const subID = this.values[SubscriptionFields.SubSelect_Name].value;
+            this.addTriggerField('id', subID);
+        }
+        this.addTriggerField('title', this.getTriggerTitle());
+        this.addTriggerField('description', this.getTriggerDescription());
+        this.addTriggerField('actions', this.getTriggerActions());
+        this.addTriggerField('conditions', this.getTriggerConditions());
+
+        return {trigger: this.trigger};
+    }
+
+    getTriggerActions(): any[] {
+        const actions = [
+            {
+                field: TriggerFields.ActionField,
+                value: [
+                    TriggerFields.TargetID,
+                    this.getJSONDataFields(),
+                ],
+            },
+        ];
+        return actions;
+    }
+
+    // getJSONDataFields constructs the object text string for a trigger
+    getJSONDataFields(): string {
+        const channelID = this.values[SubscriptionFields.ChannelPickerSelect_Name].value;
+        const pairs = TriggerFields.ActionValuePairs;
+        pairs[TriggerFields.ChannelIDKey] = channelID;
+
+        const objStrings = Object.entries(TriggerFields.ActionValuePairs).map((pair) => {
```

Comment:
Of course this is what I should have done instead of constructing myself. And I should have searched for a package that deals with JSON data. 

Done.